8.0 partner relations v8api rewind

NL66278 opened the pull request:
Hello @hbrunn . Here my merge request. Changes are quite extensive. But there was much left over from old 7.0 api that was no longer needed. And I tried to eliminate all the duplication by not having views for both res.partner.relation and res.partner.relation.all, but only interact with user through that last model.

Before merge you will need to rebase on latest 8.0, because of the removal of en.po.

NL66278 committed:
[FIX] Style improvements according to review.

NL66278 committed:
[FIX] Replace deprecated fnct_inv by reverse.

NL66278 committed:
[FIX] Use v8 no_create on field options.

NL66278 committed:
[FIX] Correct handling of create/write through left/right_partner_id.

NL66278 committed:
[FIX] Finally solve problem with wrongly displayed created relation.

NL66278 committed:
[FIX] Try to fix domain selection for partner relations.

NL66278 committed:
[FIX] Remove unneeded left/right_contact_type.

NL66278 committed:
[FIX] Remove deprecated/not used action and code.

NL66278 committed:
[ENH] Maintain standard relations just from left side.

NL66278 committed:
[FIX] Move vals correction to virtual relations.

NL66278 committed:
[FIX] Do away with unused contact_type.

NL66278 committed:
[ENH] Put all user interaction on res.partner.relation.all.

NL66278 committed:
[ENH] Improve setting domain depending on input.

NL66278 committed:
[ENH] Active relation type now computed.

Also remove deprecated ui functions from model and tests for res.partner.relation.

NL66278 committed:
[ENH] Get is_inverse straight from type_selection_id.

NL66278 committed:
[ENH] Streamlined name_get().

NL66278 committed:
[ENH] Streamlined name_search().

NL66278 committed:
[ENH] No more get_type_from_selection_id().

NL66278 committed:
[ENH] Wrong computation of relation_count.

NL66278 committed:
[ENH] Changed 'former employee' to 'employee' in demo data.

Wether an employee is an actual, a former or a future employee depends on date_start and date_end of
the relation.

NL66278 committed:
[ENH] use lowercase for name and inverse_name in demo data.

NL66278 committed:
[i18] New pot file and updated dutch translations.

NL66278 committed:
[ENH] No context manipulation and read/write override.

NL66278 committed:
[FIX] All python modules pylint 10.0 score.

Also use relation_id on res.partner.relation.all to operate on underlying object.

NL66278 committed:
[FIX] Adopt tests to changes in models.

NL66278 committed:
[FIX] Symmetric relations have left and right same type.

hbrunn commented:
after rebasing on 8.0, this branch conflicts again, so please rebase yours too

Reviewing this is going to be very tiresome because you mixed whitespace and formatting changes with code changes, resulting in a diff that simply replaces everything. Any chance to reorder your commits so that the reviewer doesn't need to read every single line closely?

NL66278 commented:
@hbrunn I rebases again, conflicts have been solved.

I tried to have each commit cover one specific area. If I needed multiple commits to get something right, I merged them afterwards. I admit that some commits (like the last one) contain both formatting changes and functional changes - but only in the same area. For review it might be best to look at the end result, rather than to the individual changes.

hbrunn commented on partner_relations/models/res_partner_relation_type_selection.py:
```diff
@@ -1,149 +1,128 @@
-from .res_partner import PADDING
+
+
+PADDING = 10
```
I think this is going to bite us if we ever want/need to change this. Why don't you import it from partner_relation_all? Given you were busy with this anyways, how about renaming the variable (technically, this is not a padding but a multiplicator)? When doing my corrections, I was tempted to rename it then already, but decided against it for the sake of small diffs.
We can also consider changing the value to 2 in order to save id space - now, this module can only handle ids in the range of MAX_INT/10, while MAX_INT/2 would be possible. Given it's a view, no migration should be needed.

NL66278 replied:
The padding (or multiplication) of res_partner_relation_type_selection and res_partner_relation_all are not completely dependent of each other, and there is no intrinsic reason they should be the same. For instance I could have excluded the symmetric types from both the record type 'a' and 'b' records and have a separate UNION to include those.

Changing the value to 2 will work because now we have two files (actually two times the same file), so for instance the normal types will then always be even and the inverse relations always odd. I adapted this technique from a case where there was a union of 3 files. I called it PADDING because basically all normal types get a '0' stuffed on the right side, and all inverse types a '1'. Using multiplication just was the easiest way to do this.

I think we will be hugely succesfull with our system if we hit the MAXINT/10 limit. Although I will think Odoo performance problems might actually somewhat hinder reaching that target.

I have no problem with changing the names of the variables, or their values. Bit I think it would be logically wrong to include the one for the other.

hbrunn replied:
yes, thanks for elaborating, they have nothing to do with each other. A quick glance makes me think that there are also no instance where we'd need a PADDING1 and PADDING2 to differenciate between both, so let's keep it the way it is. The other point was indeed quite theoretical and only meant for the case you touch this anyways. Then let's figure out my two general questions and then I'd merge immediately

hbrunn commented:
great simplifications!

Two concerns:
- you made the column `this_partner_id` visible in the tree view for the all relations model. Is it correct that this will cause the current partner to always be visible (and editable) in the list of relations of some specific partner? And if so, why do we want this? Seems a bit confusing for the user to me. My proposal would be to make the invisible attribute react to some context key which is set by the button, but not the menu entry
- the menu entry showing all relations will show both sides I assume. This might or might not be a good thing, I'm not sure. What's your opinion about this? I'd be inclined to add a domain there enforcing one side of the relation and add a primary tree view that changes the labels to 'Left' and 'Right', which the menu action pulls, but not the button

I can't have a look what runbot says, but I assume you ran the tests and they are successful. Do we still have around 90% coverage on all the files?

And github does a great job with side by side diffs, reviewing was a lot less horrible than I expected.

NL66278 commented:
I made the this_partner_id visible because I noticed that when you moved from a partner to the relations you could just drop the domain on this_partner_id, and you would be left with a list of all partners, but the this_partner_id side not shown. This was very confusing. Then I looked for a way to hide the this_partner_id column dependend on wether or not the this_partner_id was still in the selection, but I could not easily find a way to do this. I you have a suggestion on how to solve this I would be gratefull.

The menu entry is showing each relation indeed from two sides. I added quickfilters to show only the one side or the other. We might consider having a default filter to show only from left to right, but on the other hand, creating a relation from partner A tot partner B, does create a relation from B to A. Apart from that suppose we only show the left relations (left_partner_id = this_partner_id), and the user selects an inverse relation to create a new connection. After saving the new relation this relation which is right to left will be shown in a view that for the rest only contains left to right relations. This might be more easy to understand for the user if he/she consiously limited the list to left relations (and newly created ones). 

I did ran the test. It took some changes to make them all succesfull again, but now they are. By the way I added the 'ref'  field to the partner-creation in test, because I got a lot of warnings about deprecated methods used in the base_partner_sequence module while running the test. Although this is an error in the base_partner_sequence module, adding ref was an easy way to surpress the WARNING' s.

hbrunn commented:
I don't exactly understand the problem with `this_partner_id`, but will merge anyways. We can fix that later if there's something to fix in the first place

hbrunn commented:
@NL66278 you missed a few lint issues: https://travis-ci.org/OCA/partner-contact/jobs/156198865#L431

NL66278 commented:
Working on solution now. I tested with pyflakes, but flake8 seems to be more critical. Also pylint en flake8 are disagreeing in some cases on where closing bracket should be placed. I will try to get everything confirming to both pylint and flake8. There are also (non fatal) complaints about @api.one and about write and create in res_partner_relation_all not calling super (which is intended and unavoidable).
